src/reducers/weather_api_reducer.js

```diff
@@ -0,0 +1,17 @@
+import  { RECEIVE_FORECAST, RECEIVE_FORECAST_ERRORS, RECEIVE_CURRENT_WEATHER } from '../actions/weather_api_actions';
+
+const weatherReducer = (state=[], action) => {
+    Object.freeze(state);
```

Comment:
Why's this necessary? Are you breaking any redux patterns by adding this line?

Reply:
Redux does not protect state shape, so this is just a way to make sure the state form doesn't change when the store updates, therefore preventing any bugs. I want to make sure I'm not mutating it in any way.. Perhaps it wouldn't be needed in production; rather, just to prevent errors during development.